src/raft_baselines/classifiers/transformers_causal_lm_classifier.py

```diff
@@ -0,0 +1,78 @@
+from logging import Formatter
+from typing import Dict, Optional, List, Mapping
+
+import numpy as np
+import datasets
+import torch
+from transformers import AutoModelForCausalLM
+from sentence_transformers import util
+
+from raft_baselines.classifiers.in_context_classifier import InContextClassifier
+from raft_baselines.utils.tokenizers import TransformersTokenizer
+from raft_baselines.utils.embedders import SentenceTransformersEmbedder
+
+
+class TransformersCausalLMClassifier(InContextClassifier):
```

Comment:
This looks really clean - thanks for implementing it! 

One thing that would help my understanding is a small code snippet (or maybe a script?) that explains the steps one has to take to go from the RAFT dataset to a set of predictions, e.g. I'm thinking of something like

```python
from datasets import load_dataset
from raft_baselines import TransformersCausalLMClassifier

raft_dset = load_dataset("ought/raft", "banking_77")
clf = TransformersCausalLMClassifier()
# what do I do next?
```

Reply:
[raft_predict.py](https://github.com/oughtinc/raft-baselines/blob/hf-hub-generative-baselines/src/raft_baselines/scripts/raft_predict.py) is the script which runs classifiers on predictions; the forthcoming starter kit notebook should contain the code for this as well